test/tests_for_utils/application_info_test.py

```diff
@@ -0,0 +1,193 @@
+        a,
+        False,
+        IncorrectParamsException(
+            "Expected exactly one '.' in method ID 'mod.innermod.method'"
+        ),
+    )
+    _app_info_init_fail(
+        "    .  meth",
+        a,
+        False,
+        IncorrectParamsException(
+            "Missing input parameter: module portion of method ID"
+        ),
+    )
+    _app_info_init_fail(
+        " mod   .  ",
+        a,
+        False,
+        IncorrectParamsException(
+            "Missing input parameter: method portion of method ID"
+        ),
+    )
+
+    _app_info_init_fail(
+        m,
+        None,
+        False,
+        IncorrectParamsException("Missing input parameter: application ID"),
+    )
+    _app_info_init_fail(
+        m,
+        "   \t    ",
+        False,
+        IncorrectParamsException("Missing input parameter: application ID"),
+    )
+    _app_info_init_fail(
+        m,
+        "mod / meth.bak ",
+        False,
+        IncorrectParamsException(
+            "Application ID 'mod / meth.bak' has both '/' and '.' separators"
+        ),
+    )
+    _app_info_init_fail(
+        m,
+        "mod / meth / bak ",
+        False,
+        IncorrectParamsException(
+            "Expected exactly one '/' in application ID 'mod / meth / bak'"
+        ),
+    )
+    _app_info_init_fail(
+        m,
+        "mod.meth",
+        True,
+        IncorrectParamsException("Application ID 'mod.meth' contains a '.'"),
+    )
+    _app_info_init_fail(
+        m,
+        "mod.meth.anothermeth",
+        False,
+        IncorrectParamsException(
+            "Expected exactly one '.' in application ID 'mod.meth.anothermeth'"
+        ),
+    )
+
+    _app_info_init_fail(
+        "mod.meth",
+        "  mod2  /meth",
+        True,
+        IncorrectParamsException(
+            "Application module 'mod2' must equal method module 'mod'"
+        ),
+    )
+
+    _app_info_init_fail(
+        m,
+        "mod/",
+        False,
+        IncorrectParamsException(
+            "Missing input parameter: application portion of application ID"
+        ),
+    )
+    _app_info_init_fail(
+        m,
+        "/meth",
+        False,
+        IncorrectParamsException(
+            "Missing input parameter: module portion of application ID"
+        ),
+    )
+    _app_info_init_fail(
+        m,
+        "mod.   ",
+        False,
+        IncorrectParamsException(
+            "Missing input parameter: application portion of application ID"
+        ),
+    )
+    _app_info_init_fail(
+        m,
+        "   .meth",
+        False,
+        IncorrectParamsException(
+            "Missing input parameter: module portion of application ID"
+        ),
+    )
+
+
+def _app_info_init_fail(meth, app, strict, expected):
+    with raises(Exception) as got:
+        AppInfo(meth, app, strict)
+    assert_exception_correct(got.value, expected)
+
+
+def test_equals():
+    assert AppInfo("m.n", "m/p") == AppInfo("m.n", "m/p")
```

Comment:
how about a test where one of the comparators is a different class?

Reply:
Is this not enough? https://github.com/kbase/execution_engine2/pull/329/files#diff-14dda797937003df3d62117e81a114cd23c0c2eb185a2be7e184b246a33226edR180

Reply:
I was thinking of creating an instance of a different class, but that also works.